### PR TITLE
fix panic on --label-add

### LINKED ISSUE #25022: client panic on update --add-label
Received this panic while doing `docker service update test --label-add foo=bar` to a service that had no labels.

<details>
panic: assignment to entry in nil map

goroutine 1 [running]:
panic(0xaf9b40, 0xc820350760)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/docker/api/client/service.updateLabels(0xc820082ab0, 0xc820302e58)
	/go/src/github.com/docker/docker/api/client/service/update.go:259 +0x25d
github.com/docker/docker/api/client/service.updateService(0xc820082ab0, 0xc820302e48, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/update.go:147 +0x1fe
github.com/docker/docker/api/client/service.runUpdate(0xc820082510, 0xc820082ab0, 0x7fff8f14fb21, 0x5, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/update.go:68 +0x1f9
github.com/docker/docker/api/client/service.newUpdateCommand.func1(0xc820319d40, 0xc82028f2c0, 0x1, 0x5, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/update.go:30 +0x6e
github.com/spf13/cobra.(*Command).execute(0xc820319d40, 0xc82028f270, 0x5, 0x5, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:593 +0x6fe
github.com/spf13/cobra.(*Command).ExecuteC(0xc8200858c0, 0xc820319d40, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:683 +0x55c
github.com/spf13/cobra.(*Command).Execute(0xc8200858c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:642 +0x2d
github.com/docker/docker/cli/cobraadaptor.CobraAdaptor.run(0xc8200858c0, 0xc820082510, 0x7fff8f14fb12, 0x7, 0xc82000a120, 0x6, 0x6, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cobraadaptor/adaptor.go:120 +0x25f
github.com/docker/docker/cli/cobraadaptor.CobraAdaptor.Command.func1(0xc82000a120, 0x6, 0x6, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cobraadaptor/adaptor.go:128 +0x8c
github.com/docker/docker/cli.(*Cli).Run(0xc82033e510, 0xc82000a110, 0x7, 0x7, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cli.go:81 +0x34b
main.main()
	/go/src/github.com/docker/docker/cmd/docker/docker.go:72 +0x599
</details>

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes #25022

### CODE REVIEW

**Comment by LK4D4** [comment]
@cpuguy83 timeout in tests. I suppose it's unrelated.

File: integration-cli/docker_cli_service_update_test.go [hunk 1]
```diff
@@ -43,3 +43,44 @@ func (s *DockerSwarmSuite) TestServiceUpdatePort(c *check.C) {
...
+	c.Assert(service.Spec.Labels, checker.HasLen, 0)
+	c.Assert(service.Spec.Labels["foo"], checker.Equals, "")
+
+	// now make sure we can add again
+	out, err = d.Cmd("service", "update", "test", "--label-add", "foo=bar")
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	service = d.getService(c, "test")
+	c.Assert(service.Spec.Labels, checker.HasLen, 1)
+	c.Assert(service.Spec.Labels["foo"], checker.Equals, "bar")
+}
```
Comment (dnephin) [inline integration-cli/docker_cli_service_update_test.go]: There are unit tests for this, I think it should be fine to test this with another unit test.

File: api/client/service/update.go [hunk 2]
```diff
@@ -249,7 +249,7 @@ func updatePlacement(flags *pflag.FlagSet, placement *swarm.Placement) {
 
 func updateLabels(flags *pflag.FlagSet, field *map[string]string) {
 	if flags.Changed(flagLabelAdd) {
-		if field == nil {
+		if *field == nil {
```
Comment (dnephin) [inline api/client/service/update.go]: Do we need this same fix below in `if field != nil` ?

If someone tries to remove a label from a service that has no labels yet.

**Comment by dnephin** [comment]
Thanks for the fix!

File: integration-cli/docker_cli_service_update_test.go (hunk 1 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_service_update_test.go]: We still need to test the rest as an integration test, this isn't really adding any time to the test, it isn't going to break later... I don't think it hurts to have it here.

File: api/client/service/update.go (hunk 2 above)
Comment (cpuguy83) [inline api/client/service/update.go]: Yep, fixed it.

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**